bumble/profiles/vocs.py

```diff
@@ -0,0 +1,240 @@
+            ),
+            value=gatt.CharacteristicValue(
+                read=self._on_read_audio_location,
+                write=self._on_write_audio_location,
+            ),
+        )
+        characteristics.append(self.audio_location)
+
+        self.volume_offset_control_point = gatt.Characteristic(
+            uuid=gatt.GATT_VOLUME_OFFSET_CONTROL_POINT_CHARACTERISTIC,
+            properties=gatt.Characteristic.Properties.WRITE,
+            permissions=gatt.Characteristic.Permissions.WRITE_REQUIRES_ENCRYPTION,
+            value=gatt.CharacteristicValue(
+                write=self._on_write_volume_offset_control_point
+            ),
+        )
+        characteristics.append(self.volume_offset_control_point)
+
+        self.audio_output_description = gatt.Characteristic(
+            uuid=gatt.GATT_AUDIO_OUTPUT_DESCRIPTION_CHARACTERISTIC,
+            properties=(
+                gatt.Characteristic.Properties.READ
+                | gatt.Characteristic.Properties.NOTIFY
+                | gatt.Characteristic.Properties.WRITE_WITHOUT_RESPONSE
+            ),
+            permissions=(
+                gatt.Characteristic.Permissions.READ_REQUIRES_ENCRYPTION
+                | gatt.Characteristic.Permissions.WRITE_REQUIRES_ENCRYPTION
+            ),
+            value=gatt.CharacteristicValue(
+                read=self._on_read_audio_output_description,
+                write=self._on_write_audio_output_description,
+            ),
+        )
+        characteristics.append(self.audio_output_description)
+
+        super().__init__(characteristics=characteristics, primary=False)
+
+    @property
+    def volume_offset_state_bytes(self) -> bytes:
+        return struct.pack('<hB', self.volume_offset, self.change_counter)
+
+    @volume_offset_state_bytes.setter
+    def volume_offset_state_bytes(self, new_value: bytes) -> None:
+        self.volume_offset, self.change_counter = struct.unpack('<hB', new_value)
+
+    def _on_read_volume_offset_state(
+        self, _connection: Optional[device.Connection]
+    ) -> bytes:
+        return self.volume_offset_state_bytes
+
+    @property
+    def audio_location_bytes(self) -> bytes:
+        return struct.pack('<I', self.audio_channel_location)
+
+    @audio_location_bytes.setter
+    def audio_location_bytes(self, new_value: bytes) -> None:
+        self.audio_channel_location = struct.unpack('<I', new_value)[0]
+
+    def _on_read_audio_location(
+        self, _connection: Optional[device.Connection]
+    ) -> bytes:
+        return self.audio_location_bytes
+
+    async def _on_write_audio_location(
+        self, connection: Optional[device.Connection], value: bytes
+    ) -> None:
+        self.audio_location_bytes = value
+        await connection.device.notify_subscribers(
+            attribute=self.audio_channel_location,
+            value=self.audio_location_bytes,
+        )
+        self.emit('audio_location', self.audio_location)
```

Comment:
Typically, it is preferable to not repeat in the event the changed state if that state already exists as a property of the emitter.
Here, for instance (the same idea applies to the other placed in this PR where events are emitted), a listener can access `audio_location` from the emitter, so there's no need to include that in the event.
a listener would do:
`service.on('audio_location_change', ...)`
and in the event handler, retrieve that actual values as `service.audio_location`.

Reply:
Thank you for the explanation. I removed these unnecessary properties from the events.

Reply:
In the current version of the service I was not sure how to handle events, so I removed them completely. If they are necessary, could you please give me some example?